<launch>

  <!--
    
    Ground truth translation from /map (OptiTrack) to UAV's baselink
    /map -> /DJI/base_link_no_orientation

    Rotation of UAV's baselink to North
    /DJI/base_link_no_orientation -> /DJI/base_link_no_orientation_north_corrected 

    Absolute orientation using /dji_sdk/attitude_quaternion of UAV's baselink translation (north corrected)
    /DJI/base_link_no_orientation_north_corrected -> /DJI/base_link_estimated

    now GT from OptiTrack is very close to /DJI/base_link_estimated
    /DJI/base_link ~~~ /DJI/base_link_estimated

  -->

  <group ns="dji_sdk">
   
    <node name="att_quat_to_transform" pkg="att_quat_to_transform" type="att_quat_to_transform.py" respawn="true" output="screen">
      <param name="baselink_translation_frame_id" type="str" value="/DJI/base_link_no_orientation_north_corrected" />
      <param name="estimated_baselink_frame_id"   type="str" value="/DJI/base_link_estimated" />
    </node>

    <node name="strip_orientation_from_pose" pkg="att_quat_to_transform" type="strip_orientation_from_pose.py" respawn="true" output="screen">
      <param name="map_frame_id" type="str" value="map" />
      <param name="baselink_translation_frame_id" type="str" value="/DJI/base_link_no_orientation" />
    </node>

    <!-- Rotate OptiTrack frame to North -->
    <node name="north_correction" pkg="tf" type="static_transform_publisher" respawn="true" output="screen" 
      args="0 0 0 0.776 0 0 /DJI/base_link_no_orientation /DJI/base_link_no_orientation_north_corrected 100">
    </node>
  </group>
</launch>
